scipy/spatial/transform/_rotation.pyx

```diff
@@ -2124,13 +2171,93 @@ cdef class Rotation:
         cdef double[:] angles = _empty1(num_rotations)
 
         for ind in range(num_rotations):
-            angles[ind] = 2 * atan2(_norm3(quat[ind, :3]), abs(quat[ind, 3]))
+            angles[ind] = _quat_magnitude(quat[ind])
 
         if self._single:
             return angles[0]
         else:
             return np.asarray(angles)
 
+    @cython.embedsignature(True)
+    @cython.boundscheck(False)
+    @cython.wraparound(False)
+    def angle_to(Rotation self, Rotation other):
+        """Get the smallest angular distance(s) to another rotation.
+
+        For two rotations ``p`` and ``q``, the angular distance
+        ``p.angle_to(q)`` is equivalent to calculating
+        ``(q * p.inv()).magnitude()``.
+
+        Parameters
+        ----------
+        other : `Rotation` instance
+            Object containing the rotations to measure the angle to.
+
+        Returns
+        -------
+        angle : ndarray or float
+            Angle(s) in radians, float if object contains a single rotation
+            and ndarray if object contains multiple rotations. The angle
+            will always be in the range [0, pi].
+
+        Examples
+        --------
+        >>> from scipy.spatial.transform import Rotation as R
+        >>> import numpy as np
+        >>> p = R.from_quat([0, 0, 0, 1])
+        >>> q = R.from_quat(np.eye(4))
+        >>> p.angle_to(q)
+        array([3.14159265, 3.14159265, 3.14159265, 0.        ])
+
+        Angle to a single rotation:
+
+        >>> p.angle_to(q[0])
+        3.141592653589793
+        """
+        return self.rotation_to(other).magnitude()
+
+
+    @cython.embedsignature(True)
+    @cython.boundscheck(False)
+    @cython.wraparound(False)
+    def approx_equal(Rotation self, Rotation other, float atol=1e-12):
```

Comment:
```suggestion
    def approx_equal(Rotation self, Rotation other, double atol=1e-12):
```

`float atol` in this context is interpreted by Cython as specifying a C `float` rather than a Python `float` (which is a C `double`).